### PR TITLE
Created a resource for users

### PR DESCRIPTION
**Pull request by mdang8**
- HTML resource generated to handle users
- Users are shown a login form on the main index page
  - If a user is already logged in, then they will be shown their dashboard or feed (haven't implemented yet)
- If a user doesn't have an existing account, they can create one by clicking a link on the index page to be brought to a "Register New User" page
  - Prompted to enter an email, username, name, and password
  - Email and username have to be unique

TODO:
- Remove the ability to edit/delete users if not logged in
  - Move this to a separate page where a user can edit their own details or delete their account entirely
  - Probably will just remove the feature to view the list of all users for the final product

### CODE REVIEW

File: travelpal/lib/travelpal_web/controllers/session_controller.ex [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  alias Travelpal.Accounts
+  alias Travelpal.Accounts.User
+
+  def create(conn, %{"username" => username}) do
+    user = Accounts.get_user_by_username(username)
+
+    if user do
+      conn
+      |> put_session(:user_id, user.id)
+      |> put_flash(:info, "Welcome back #{user.username} 🙌")
```
Comment (kimberlypn) [inline travelpal/lib/travelpal_web/controllers/session_controller.ex]: Nitpick: can you add a comma after "back" and a period or exclamation mark after the username?
`"Welcome back, #{user.username}! 🙌"`

File: travelpal/lib/travelpal_web/controllers/session_controller.ex [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      conn
+      |> put_flash(:error, "Error with creating session.")
+      |> redirect(to: page_path(conn, :index))
+    end
+  end
+
+  def delete(conn, _params) do
+    conn
+    |> delete_session(:user_id)
+    |> put_flash(:info, "Logged out 👋")
```
Comment (kimberlypn) [inline travelpal/lib/travelpal_web/controllers/session_controller.ex]: Nitpick: can you add a period at the end?
`"Logged out. 👋"`

File: travelpal/lib/travelpal_web/controllers/session_controller.ex (hunk 1 above)
Comment (mdang8) [reply travelpal/lib/travelpal_web/controllers/session_controller.ex]: Fixed

File: travelpal/lib/travelpal_web/controllers/session_controller.ex (hunk 2 above)
Comment (mdang8) [reply travelpal/lib/travelpal_web/controllers/session_controller.ex]: Fixed

**Pull request merged**